src/classes/BDI_HelpTextCMT.cls

```diff
@@ -0,0 +1,249 @@
+            List<String> parts = helpText.split('\\.', 2);
+
+            this.objectName = String.isNotBlank(parts[0]) ? parts[0] : null;
+            this.fieldName = String.isNotBlank(parts[1]) ? parts[1] : null;
+
+            if (String.isNotBlank(this.objectName)) {
+                this.objectAPIName = DI_OBJECT_MAP.get(this.objectName);
+            }
+        }
+
+    }
+
+    /**
+     * Intermediary wrapper class for Data_Import_Field_Mapping__mdt; Probably scrub later and use DataImportFieldMapping directly.
+     */
+    public class MappingDetail {
+        public String dataImportObjectName;
+        public String targetObjectName;
+        public String dataImportFieldLabel;
+        public String targetFieldAPIName;
+        public String importedRecordField;
+        public String importedRecordStatusField;
+
+        public MappingDetail(String dataImportObjectName, String dataImportFieldLabel, String targetObjectName,
+                             String targetFieldAPIName, String importedRecordField, String importedRecordStatusField) {
+            this.dataImportObjectName = dataImportObjectName;
+            this.dataImportFieldLabel = dataImportFieldLabel;
+            this.targetObjectName = targetObjectName;
+            this.targetFieldAPIName = targetFieldAPIName;
+            this.importedRecordField = importedRecordField;
+            this.importedRecordStatusField = importedRecordStatusField;
+        }
+
+    }
+
+    /**
+     * @description Utility to convert a name into a valid 'DeveloperName' quality value (or unique GUID)
+     * for the CMT record
+     * @param name String
+     * @param makeUnique Boolean
+     * @return DeveloperName format string
+     */
+    private static String generateRecordName(String name, Boolean makeUnique, String objectLabel, String objectAPIName) {
+        name = name.replaceAll('[^\\w]+', '_').replaceAll('_{2,}', '_');
+
+        if (String.isNotBlank(objectLabel)) {
+            objectLabel = objectLabel.replace(' ', '_');
+            name = name.replace(objectAPIName, objectLabel);
+        }
+
+        if (name.length() > 30) {
+            String firstFifteen = name.substring(0, 15);
+            String lastFifteen = name.substring(name.length() - 15, name.length());
+            name = firstFifteen + lastFifteen;
+        }
+
+        if (makeUnique == true) {
+            // Create a GUID format unique string
+            Blob b = Crypto.GenerateAESKey(128);
+            String h = EncodingUtil.ConvertTohex(b);
+            // force everything to have a 9-char random string at the end
+            name = name.left(30).removeEnd('_') + '_' + h.left(9);
+        } else {
+            name = name.left(40);
+        }
+
+        return name;
+    }
+
+    /** @description Map of object names from the data import help text field to actual object names */
+    private static final Map<String, String> DI_OBJECT_MAP = new Map<String, String>{
```

Comment:
Even though most of our static finals are all caps this seemed a bit unusual for a collection (instead of a string or integer). But I honestly don't know why this wouldn't follow the convention of other static finals, aka essentially constants